docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,183 @@ One challenge when developing locally is that a developer might make modificatio
+
+##### Using the Gatsby preview extension
+
+We are currently in process of creating an open source package to handle most of this functionality for you. In the meantime, you will need to create a button in your CMS that does the following...
+
+1. `POST` to the preview webhook url in Gatsby Cloud
+2. Open the Content Sync waiting room
+
+The button might look something like this:
+![Open Preview Button Screenshot](https://user-images.githubusercontent.com/18426780/61498855-37e35a00-a982-11e9-8201-ab96be74f1f1.png)
+
+##### Configuration
+
+You will need to store the Content Sync URL from a given Gatsby Cloud site. This will look something like `https://gatsbyjs.com/content-sync/<siteId>`. This is often done in the CMS plugin extension configuration, this will differ from CMS to CMS depending on how they handle their plugin ecosystem.
+
+Depending on the CMS you will also need to store the preview webhook URL. This might also be stored in the plugin extension configuration, but often is stored in a separate webhooks configuration. [Find out how to get that webhook url here](https://support.gatsbyjs.com/hc/en-us/articles/360052324394-Build-and-Preview-Webhooks)
+
+NOTE: The Content Sync URL can be found in the same place as the webhook url.
+
+##### Building up the manifest id
+
+Recall that we need to create a matching manifest id in the CMS AND the Gatsby plugin. Whenever content is saved we can build up a new manifest id that will look the same as the manifest id we created in the source plugin
+
+```javascript
+const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
+```
+
+In the CMS extension, we should have access to
+
+- the project id (if the CMS uses one)
+- the content id
+- the timestamp that the content was updated at (or some other piece of data that is tied to a very specific state of saved content)
+
+##### Starting the preview build (optional)
+
+If the CMS does not handle this part automatically we will need to tell Gatsby cloud to build a preview by `POST`ing to the Gatsby Cloud preview build webhook url.
+
+##### Opening the Content Sync waiting room
+
+Once we've built a `manifestId` and `POST`ed to the preview build webhook url, we need to open a new tab/window with a modified version of the Content Sync URL. You get that by grabbing the Content Sync URL you stored in the CMS extension earlier and appending the Gatsby source plugin name and the content's `manifestId` that you just created, `https://gatsbyjs.com/content-sync/<siteId>/<sourcePluginName>/<manifestId>`.
+
+#### Useful Content Sync development tips
+
+Here are some things to keep in mind and some "gotchas" depending on how the CMS acts.
+
+- Sometimes you will need to wait to make sure you have the correct `updatedAt` timestamp as some CMS may take a second to update their backend and then wait for the change to propagate to the frontend. While others will immediately update the frontend and then propagate that to the backend.
```

Comment:
It'd be good to mention this is within the CMS, it's not 100% clear where we're waiting